mephisto/abstractions/providers/prolific/prolific_provider.py

```diff
@@ -0,0 +1,118 @@
+from mephisto.abstractions.providers.prolific.prolific_worker import ProlificWorker
+from mephisto.operations.registry import register_mephisto_abstraction
+
+if TYPE_CHECKING:
+    from mephisto.data_model.task_run import TaskRun
+    from mephisto.data_model.unit import Unit
+    from mephisto.data_model.worker import Worker
+    from mephisto.data_model.requester import Requester
+    from mephisto.data_model.agent import Agent
+    from mephisto.abstractions.blueprint import SharedTaskState
+    from omegaconf import DictConfig
+
+
+@dataclass
+class ProlificProviderArgs(ProviderArgs):
+    """Base class for arguments to configure Crowd Providers"""
+    _provider_type: str = PROVIDER_TYPE
+
+
+@register_mephisto_abstraction()
+class ProlificProvider(CrowdProvider):
+    """
+    Prolific implementation of a CrowdProvider that stores everything
+    in a local state in the class for use in tests.
+    """
+    UnitClass: ClassVar[Type["Unit"]] = ProlificUnit
+
+    RequesterClass: ClassVar[Type["Requester"]] = ProlificRequester
+
+    WorkerClass: ClassVar[Type["Worker"]] = ProlificWorker
+
+    AgentClass: ClassVar[Type["Agent"]] = ProlificAgent
+
+    ArgsClass = ProlificProviderArgs
+
+    PROVIDER_TYPE = PROVIDER_TYPE
+
+    curr_db_location: ClassVar[str]
+
+    def initialize_provider_datastore(self, storage_path: str) -> Any:
+        return ProlificDatastore(datastore_root=storage_path)
+
+    def _get_client(self, requester_name: str) -> Any:
+        """Get a Prolific client"""
+        return self.datastore.get_client_for_requester(requester_name)
+
+    def setup_resources_for_task_run(
+        self,
+        task_run: "TaskRun",
+        args: "DictConfig",
+        shared_state: "SharedTaskState",
+        server_url: str,
+    ) -> None:
+        requester = cast('ProlificRequester', task_run.get_requester())
+        task_run_id = task_run.db_id
+
+        # Set up Task Run config
+        config_dir = os.path.join(self.datastore.datastore_root, task_run_id)
+        task_args = args.task
+
+        # Find or create relevant qualifications
+        qualifications = []
+        for qualification in shared_state.qualifications:
+            applicable_providers = qualification['applicable_providers']
+
+            if applicable_providers is None or self.PROVIDER_TYPE in applicable_providers:
+                qualifications.append(qualification)
+
+        for qualification in qualifications:
+            qualification_name = qualification['qualification_name']
+
+            if self.datastore.get_qualification_mapping(qualification_name) is None:
+                qualification['id'] = requester.create_new_qualification(qualification_name)
+
+        if hasattr(shared_state, 'prolific_specific_qualifications'):
+            # TODO(OWN) standardize provider-specific qualifications
+            qualifications += shared_state.prolific_specific_qualifications
```

Comment:
As a note, qualifications are likely going to be the most unusual part of Prolific for Mephisto. While _we're_ happy to assign qualifications to users, and then make those into requirements or exclusions, Prolific has no way to directly do this.

They _do_ however have an allowlist/blocklist feature, and the allowlist and blocklist for a task can be updated dynamically. As such _we'll_ have to do the marginal qualification 'calculation's whenever one of the targeted qualifications has a change. I'm open to a number of directions, but I think the easiest would be:
1. Store the target qualifications that may affect a task in the `ProlificDatastore`
2. On a `Worker.<grant/revoke>_crowd_qualification(...)` call, check the datastore, and push updates to relevant allow/blocklists for active tasks.